lib/active_merchant/billing/gateways/xpay.rb

```diff
@@ -112,23 +197,23 @@ def build_request_url(action, id = nil)
       end
 
       def success_from(response)
-        response == 'SUCCESS'
+        response.dig('operation', 'operationResult') == 'PENDING' || response.dig('operation', 'operationResult') == 'FAILED' || response.dig('operation', 'operationResult') == 'AUTHORIZED'
       end
 
-      def message_from(succeeded, response)
-        if succeeded
-          'Succeeded'
+      def message_from(response)
+        if response.include? 'errors'
+          response['errors']
         else
-          response.dig('errors') unless response
+          response.dig('operation', 'operationResult')
         end
       end
 
       def authorization_from(response)
-        response.dig('latest_payment_attempt', 'payment_intent_id') unless response
+        response.dig('operation', 'operationId') unless response
       end
 
       def error_code_from(response)
-        response['provider_original_response_code'] || response['code'] unless success_from(response)
+        response.dig('errors')&.first&.dig('code') unless success_from(response)
```

Comment:
💬 Comment ❓ Question:

Comment => dig can handle this logic directly:

```ruby
def error_code_from(response)
  response.dig('errors', 0, 'code')
end
```

Question => checking just the first error `0` is because of the API response restructure or we are just going to show just the first message ?

Reply:
Just the first message, so far the tests only shows one error.